### PR TITLE
chore: update and simplify contribution guide

### PR DESCRIPTION
**Pull request by nija-at**
I've received feedback from some folks who would like to start
contributing to the CDK but found the contribution guide unwieldy.

Attempting to wittle down and simplify. Move content around so that
useful information is at the top.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -275,13 +311,9 @@ programs that customers could have been writing against the current version
 of the CDK, that will no longer "work correctly" with the proposed new
 version of the CDK.
 
-Breaking changes are not allowed in *stable* libraries¹. They are permissible
-but still *highly discouraged* in experimental libraries, and require explicit
-callouts in the bodies of Pull Requests that introduce them.
-
-> ¹) Note that starting in version 2 of the CDK, the majority of library code will be
```
Comment (rix0rrr) [inline CONTRIBUTING.md:282]: Elad explicitly requested I was super explicit about this, which is why it's here.

Do you think it's not worthwhile? (I mean what the section says is sorta implied by the section before it, but again, explicitness...)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
...
+If you want to contribute a specific feature or fix you have in mind, look at active [pull
+requests](https://github.com/aws/aws-cdk/pulls) to see if someone else is already working on it. If not, you can start
+contributing your changes.
 
-Once the design is finalized, you can re-purpose this PR for the implementation,
-or open a new PR to that end.
+On the other hand, if you are here looking for an issue to work on, explore our [backlog of
+issues](https://github.com/aws/aws-cdk/issues) and find something that piques your interest. We have labeled all of our
+issues for easy filtration.
+If you are looking for your first contribution, the 'good-first-issue' label will be of help.
```
Comment (jogold) [inline CONTRIBUTING.md:188]:
```suggestion
If you are looking for your first contribution, the ['good first issue' label](https://github.com/aws/aws-cdk/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) will be of help.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
...
+You may also set up your local development environment using [Gitpod](http://gitpod.io) -
+a service that allows you to spin up an in-browser Visual Studio Code-compatible editor,
+with everything set up and ready to go for CDK development.
+Just click the button below to create your private workspace:
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/aws/aws-cdk)
+
+This will start a new Gitpod workspace, and immediately kick off a build of the CDK code.
+Once it's done (it takes around an hour, unfortunately), you can work on any package that you want to modify,
+as described in ['Quick Iteration'](#quick-iteration) below.
```
Comment (skinny85) [inline CONTRIBUTING.md:161]: Fortunately, we now use pre-builds, so the one-hour wait time is no more! We can remove that part.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
...
-you need to have the following SDKs and tools locally:
+The following tools need to be installed on your system prior to installing the CDK:
 
 - [Node.js >= 10.13.0](https://nodejs.org/download/release/latest-v10.x/)
   - We recommend using a version in [Active LTS](https://nodejs.org/en/about/releases/)
   - ⚠️ versions `13.0.0` to `13.6.0` are not supported due to compatibility issues with our dependencies.
 - [Yarn >= 1.19.1, < 2](https://yarnpkg.com/lang/en/docs/install)
-- [Java >= OpenJDK 8, 11, 14](https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html)
-- [Apache Maven >= 3.6.0, < 4.0](http://maven.apache.org/install.html)
-- [.NET Core SDK 3.1.x](https://www.microsoft.com/net/download)
```
Comment (skinny85) [inline CONTRIBUTING.md:84]: Unfortunately, .NET Core is required (to run the test cases inside the `aws-cdk` package).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
...
-- [Python >= 3.6.5, < 4.0](https://www.python.org/downloads/release/python-365/)
-- [Docker >= 19.03](https://docs.docker.com/get-docker/)
```
Comment (skinny85) [inline CONTRIBUTING.md:86]: Unfortunately, Docker is required (to run many tests in the `@aws-cdk/aws-lambda-xyz` packages, but also to build the `@aws-cdk/lambda-layer-awscli` and `@aws-cdk/lambda-layer-kubectl` packages).

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
...
+The following tools need to be installed on your system prior to installing the CDK:
 
 - [Node.js >= 10.13.0](https://nodejs.org/download/release/latest-v10.x/)
   - We recommend using a version in [Active LTS](https://nodejs.org/en/about/releases/)
   - ⚠️ versions `13.0.0` to `13.6.0` are not supported due to compatibility issues with our dependencies.
 - [Yarn >= 1.19.1, < 2](https://yarnpkg.com/lang/en/docs/install)
-- [Java >= OpenJDK 8, 11, 14](https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html)
-- [Apache Maven >= 3.6.0, < 4.0](http://maven.apache.org/install.html)
-- [.NET Core SDK 3.1.x](https://www.microsoft.com/net/download)
-- [Python >= 3.6.5, < 4.0](https://www.python.org/downloads/release/python-365/)
```
Comment (skinny85) [inline CONTRIBUTING.md:85]: I believe Python is required (to build modules that use Python-based custom resources, like `@aws-cdk/aws-s3-deployment`, and `@aws-cdk/aws-s3-notifications`.

**Review by skinny85 — DISMISSED** [review]
Looks good! Some minor comments.

File: CONTRIBUTING.md (hunk 6 above)
Comment (nija-at) [reply CONTRIBUTING.md:85]: Good catches. Added them back.

File: CONTRIBUTING.md (hunk 2 above)
Comment (nija-at) [reply CONTRIBUTING.md:188]: Thanks! Fixed.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**